lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyCompletionParticipant.java

```diff
@@ -53,12 +54,33 @@ public void onAttributeValue(String valuePrefix, ICompletionRequest request, ICo
         Properties variableProps = SettingsService.getInstance()
                 .getVariablesForServerXml(request.getXMLDocument()
                         .getDocumentURI());
-        String variableName = valuePrefix.replace("$", "")
-                .replace("{", "")
+        //getting all existing variables in current completion prefix string
+        List<VariableLoc> variables = LibertyUtils.getVariablesFromTextContent(valuePrefix);
+        // value prefix will contain all existing attribute value, till the cursor
+        String variableName = valuePrefix.replace("${", "")
                 .replace("}", "");
-        variableProps.entrySet().stream().filter(it -> it.getKey().toString().toLowerCase().contains(variableName.toLowerCase()))
+        StringBuilder prefixVariables = new StringBuilder();
+        String replacePrefix = "";
+        for (VariableLoc variableLoc : variables) {
+            prefixVariables.append(String.format("${%s}", variableLoc.getValue()));
+        }
+        if (!variables.isEmpty()) {
+            // find last variable last character and consider new variable completion from that location
```

Comment:
Do we know the actual position of the cursor when they triggered completion?

Reply:
In the onAttributeValue() method, first argument is the prefix variable. ie, if user has entered ${abc|, prefix will contain ${abc